### ISSUE TITLE
Add Cammus C12 RPM Led and Lcd support

### ISSUE THREAD

**Issue by spikerguy** [Spacefreak18/monocoque#8]
@KaJe5380

```
Bus 003 Device 009: ID 3416:1023 Cammus_C12_HUB_V021 Cammus_C12_HUB_V009 Buttons
Bus 003 Device 010: ID 3416:0302 CAMMUS CAMMUS C12 Base
```

can you try this

`printf '\xfc\x05\x01\xff\x03' > /dev/hidraw5`

Need to run it as root, given that hidraw5 is still attached to Cammus_C12_HUB_V021 Cammus_C12_HUB_V009 Buttons

if hidraw number changes then change it in the command.

let me know if the rpm led and lcd output something.

Regarding monocoque we are working on getting more games and devices supported.

**Comment by spikerguy** [comment]
This will not work. So we will have to look into the usb packets to understand how it is done on C12.

**Comment by spikerguy** [comment]
@Spacefreak18 

```
### Lcd and rpm led logics Cammus C12 Hub
VID:3416
PID:1023

## RPM Led and LCD update
\xfa\xfb\xd4\x25\x01\x02\x4

Bit 1 ,2 and 3 = secret to update rpm led and lcd
bit 4 = rpm led
bit 5 and 6 = speed
bit 7 = gear

### Set LCD display data logic

fafb0602 it for speed in kmph

Bit 1,2 and 3 = secret to set lcd display logic
Bit 4 = 1, 2 or 3
1 = Display only gear
2 = Display speed in kmph
3 = Display speed in mph
Example
Fafb0601 it for lcd show only gear and ignores speed
Fafb0603 is  for speed in miles ph
```

Please add when you get time

**Comment by spikerguy** [comment]
Other functions can also be added 

```
### Set LCD display data logic

fafb0602 it for speed in kmph

Bit 1,2 and 3 = secret to set lcd display logic
Bit 4 = 1, 2 or 3
1 = Display only gear
2 = Display speed in kmph
3 = Display speed in mph
Example
Fafb0601 it for lcd show only gear and ignores speed
Fafb0603 is  for speed in miles ph

### LCD Brightness
Xfa\xfb\x01\06
Bit 1, 2 and 3 + secret to change lcd Brightness
Bit 4 = 1 to 8 where 6 of 8 is set

### Manual led control - Need more testing for color and Brightness
xfa\xfb\x02\x01\x10
Bit 1,2 and 3 = secret to manually change led colour or Brightness
Bit 4 = led number
bit 5 = colour
```

**Comment by Spacefreak18** [comment]
> @Spacefreak18
> 
> ```
> ### Lcd and rpm led logics Cammus C12 Hub
> VID:3416
> PID:1023
> 
> ## RPM Led and LCD update
> \xfa\xfb\xd4\x25\x01\x02\x4
> 
> Bit 1 ,2 and 3 = secret to update rpm led and lcd
> bit 4 = rpm led
> bit 5 and 6 = speed
> bit 7 = gear
> 
> ### Set LCD display data logic
> 
> fafb0602 it for speed in kmph
> 
> Bit 1,2 and 3 = secret to set lcd display logic
> Bit 4 = 1, 2 or 3
> 1 = Display only gear
> 2 = Display speed in kmph
> 3 = Display speed in mph
> Example
> Fafb0601 it for lcd show only gear and ignores speed
> Fafb0603 is  for speed in miles ph
> ```
> 
> Please add when you get time

are you sure about some of this?

for example 
## RPM Led and LCD update
\xfa\xfb\xd4\x25\x01\x02\x4

if byte 4 is the number of leds to light, the hex 25, is decimal 35, 37 leds?

if byte 7 is gear, and if that's hex 40, 64th gear?

**Comment by spikerguy** [comment]
rpm led data is sent out in percentage in hex 
x3c = 60%
x43 = 67%
x44 = 68%


gear data is plan decimals  x04 = 4th gear and x01 = 1st gear

I hope everything is clear for rpm led, lce for speed and gear :D

**Comment by Spacefreak18** [comment]
should be, i figured it was x04 but i wanted to be sure after looking at the rest of it

**Comment by spikerguy** [comment]
This can be closed as it is fixed in this commit a81004418dda6e7f59cd560a546ff55adb670386.

Thank you so much for this project and the quick support.

Can now fully utilize Cammus C5 and C12 telemetry IO.
